sigstore/_internal/rekor/client_v2.py

```diff
@@ -0,0 +1,187 @@
+
+        integrated_entry = resp.json()
+        _logger.debug(f"integrated: {integrated_entry}")
+        return LogEntry._from_dict_rekor(integrated_entry)
+
+    @classmethod
+    def _build_hashed_rekord_create_entry_request(
+        cls,
+        artifact_hashed_input: Hashed,
+        artifact_signature: bytes,
+        signining_certificate: Certificate,
+    ) -> v2.CreateEntryRequest:
+        return v2.CreateEntryRequest(
+            hashed_rekord_request_v0_0_2=v2.HashedRekordRequestV002(
+                digest=artifact_hashed_input.digest,
+                signature=v2.Signature(
+                    content=artifact_signature,
+                    verifier=v2.Verifier(
+                        public_key=v2.PublicKey(
+                            raw_bytes=signining_certificate.public_key().public_bytes(
+                                encoding=serialization.Encoding.DER,
+                                format=serialization.PublicFormat.SubjectPublicKeyInfo,
+                            )
+                        ),
+                        key_details=DEFAULT_KEY_DETAILS,  # type: ignore[arg-type]
+                    ),
+                ),
+            )
+        )
+
+    @classmethod
+    def _build_dsse_create_entry_request(
+        cls, envelope: Envelope, signing_certificate: Certificate
+    ) -> v2.CreateEntryRequest:
+        return v2.CreateEntryRequest(
+            dsse_request_v0_0_2=v2.DsseRequestV002(
+                envelope=v2_intoto.Envelope(
+                    payload=envelope._inner.payload,
+                    payload_type=envelope._inner.payload_type,
+                    signatures=[
+                        v2_intoto.Signature(
+                            keyid=signature.keyid,
+                            sig=signature.sig,
+                        )
+                        for signature in envelope._inner.signatures
+                    ],
+                ),
+                verifiers=[
+                    v2.Verifier(
+                        public_key=v2.PublicKey(
+                            raw_bytes=signing_certificate.public_key().public_bytes(
+                                encoding=serialization.Encoding.DER,
+                                format=serialization.PublicFormat.SubjectPublicKeyInfo,
+                            )
+                        ),
+                        key_details=DEFAULT_KEY_DETAILS,  # type: ignore[arg-type]
+                    )
+                ],
+            )
+        )
+
+    @classmethod
+    def production(cls) -> RekorV2Client:
+        """
+        Returns a `RekorV2Client` populated with the default Rekor production instance.
+        """
+        return cls(
+            DEFAULT_REKOR_URL,
+        )
+
+    @classmethod
+    def staging(cls) -> RekorV2Client:
+        """
+        Returns a `RekorV2Client` populated with the default Rekor staging instance.
+        """
+        return cls(STAGING_REKOR_URL)
```

Comment:
I think we do not need these (as long as #1363 merges): the rekor v1 client still needs them for the "fix-bundle" special case but that should not apply to this new code.

Reply:
removed.